packages/monitors/src/kpi-monitor.ts

```diff
@@ -0,0 +1,330 @@
+    );
+    const monthlyActive = accounts.filter(
+      (a) => a.lastUpdateTimestamp > nowSeconds - secondsInDay * 30
+    );
+
+    kpis.accounts = {
+      total: {
+        overall: accounts.length,
+        borrowers: accounts.filter(
+          (a) => a.hasCashDebt || a.hasPortfolioAssetDebt
+        ).length,
+      },
+      daily: {
+        overall: dailyActive.length,
+        borrowers: dailyActive.filter(
+          (a) => a.hasCashDebt || a.hasPortfolioAssetDebt
+        ).length,
+      },
+      weekly: {
+        overall: weeklyActive.length,
+        borrowers: weeklyActive.filter(
+          (a) => a.hasCashDebt || a.hasPortfolioAssetDebt
+        ).length,
+      },
+      monthly: {
+        overall: monthlyActive.length,
+        borrowers: monthlyActive.filter(
+          (a) => a.hasCashDebt || a.hasPortfolioAssetDebt
+        ).length,
+      },
+    };
+
+    Object.keys(kpis.accounts).forEach((period) => {
+      const { overall, borrowers } = kpis.accounts[period];
+      const overallSeries = {
+        metric: `kpi.accounts.${period}.overall}`,
+        tags: [`period:${period}`, `type:overall`, `chain:mainnet`],
+        points: [
+          {
+            timestamp: Math.round(new Date().getTime() / 1000),
+            value: overall,
+          },
+        ],
+      };
+      const borrowerSeries = {
+        metric: `kpi.accounts.${period}.borrowers`,
+        tags: [`period:${period}`, `type:borrowers`, `chain:mainnet`],
+        points: [
+          {
+            timestamp: Math.round(new Date().getTime() / 1000),
+            value: borrowers,
+          },
+        ],
+      };
+      series.push(overallSeries, borrowerSeries);
+    });
+  } catch (e) {
+    console.log(e);
+    await log({
+      message: (e as Error).message,
+      level: 'error',
+      chain: 'mainnet',
+      action: 'kpi.accounts',
+    });
+  }
+}
+
+function setExchangeRatesUSD(rates: ExchangeRate[]) {
+  const usdRates = rates.filter((r) => r.base === 'usd');
+  ['usd', 'btc', 'eth'].forEach((base) => {
+    const baseRates = rates.filter((r) => r.base === base);
+    baseRates.forEach((r) => {
+      if (r.base === 'usd') {
+        exchangeRatesUSD.set(r.quote, r);
+      } else {
+        const quoteRate = BigNumber.from(r.value);
+        const usdRate = usdRates.find((ur) => ur.quote === r.base);
+        if (usdRate) {
+          const rate = quoteRate
+            .mul(BigNumber.from(usdRate.value))
+            .div(BigNumber.from(10).pow(usdRate.decimals));
+          exchangeRatesUSD.set(r.quote, { ...r, value: rate });
+        }
+      }
+    });
+  });
+}
+
+const run = async ({ env }: JobOptions) => {
+  try {
+    const id = env.EXCHANGE_RATE_STORE.idFromName(
+      env.EXCHANGE_RATES_WORKER_NAME
+    );
+    console.log(`ExchangeRate id: ${id} `);
+    const stub = env.EXCHANGE_RATE_STORE.get(id);
+    const req = new Request(
+      `${env.EXCHANGE_RATE_URL}/exchange-rates?network=homestead`
```

Comment:
We don't use the "homestead" terminology anywhere in the rest of our code. We should rename homestead => mainnet everywhere.

Also the network here should be in an environment variable

Reply:
I'll handle renaming it in a separate PR.